src/main/java/de/craften/plugins/rpgplus/components/ability/AbilityComponent.java

```diff
@@ -181,6 +196,24 @@ public void onQuit(PlayerQuitEvent event) {
 		});
 	}
 
+	private static CompletableFuture<Boolean> isActive(PlayerDataStore store, Ability ability) {
+		return store.getAsync("abilitymanager." + ability.getIdentifier() + ".active").thenApply((active) -> {
+			try {
+				return Boolean.valueOf(Boolean.parseBoolean(active));
+			} catch (Exception e) {
+				return Boolean.valueOf(false);

```

Comment:
Please don't ignore exceptions! :wink: This should be re-thrown, e.g. `throw new RuntimeException("Could not check if ability is active", e);`

Reply:
Also, since you do almost everything synchronously, there isn't _really_ a need to make this asynchronous.